zyte_common_items/processors.py

```diff
@@ -59,12 +64,47 @@ def brand_processor(value: Any) -> Any:
     Other inputs are returned as is.
     """
 
-    if isinstance(value, SelectorList):
-        if len(value) == 0:
-            return None
-        value = value[0]
+    value = _handle_selectorlist(value)
+    if not isinstance(value, (Selector, HtmlElement)):
+        return value
 
-    if isinstance(value, (Selector, HtmlElement)):
-        return extract_brand_name(value, search_depth=2)
+    return extract_brand_name(value, search_depth=2)
+
+
+def price_processor(value: Any, page: Any) -> Any:
+    """Convert the data into a price string if possible.
+
+    Supported inputs are :class:`~parsel.selector.Selector`,
+    :class:`~parsel.selector.SelectorList` and :class:`~lxml.html.HtmlElement`.
+    Other inputs are returned as is.
+
+    Puts the parsed Price object into ``page._parsed_price``.
+    """
+
+    value = _handle_selectorlist(value)
+    if not isinstance(value, (Selector, HtmlElement)):
+        return value
+
+    price = extract_price(value)
+    page._parsed_price = price
+    if price.amount is None:
+        return None
+    return str(price.amount)
+
+
+def simple_price_processor(value: Any) -> Any:
+    """Convert the data into a price string if possible.
+
+    Supported inputs are :class:`~parsel.selector.Selector`,
+    :class:`~parsel.selector.SelectorList` and :class:`~lxml.html.HtmlElement`.
+    Other inputs are returned as is.
+    """
+
+    value = _handle_selectorlist(value)
+    if not isinstance(value, (Selector, HtmlElement)):
+        return value
```

Comment:
This code also repeats a few times; maybe a time to define a decorator?

```py
@only_handle_selectorlist
def simple_price_processor(value: Any) -> Any:
    price = extract_price(value)
    if price.amount is None:
        return None
    return str(price.amount)
```

Reply:
Is `value: str` a mistake here? Or what do you want the decorator to do? It looks like the decorated functions need to take `Union[Selector, HtmlElement]`.

Reply:
ah, right, it should be a selector/html element; I got confused